src/api/app/controllers/staging/staging_projects_controller.rb

```diff
@@ -20,7 +20,14 @@ def index
   end
 
   def show
-    @staging_project = @project.staging.staging_projects.find_by!(name: params[:staging_project_name])
```

Comment:
Hm, isn't that better to raise `ActiveRecord::RecordNotFound` and catch it on `application_controller` ? That's, IMO, the "OBS way" to handle it.

Reply:
It's already caught there. That's why you get a 404 and not a 500. It's just that `RecordNotFound` renders this rather strange error. So to fix it 'the OBS way', you would have to raise e.g. InvalidProjectParameters

Reply:
@coolo I'm fine with that, even before if we go DRY and use the same code doing that for `show` and `accept`, or do you think that's better to do the error handling in every single action?

Reply:
I don't care either way. I'm just saying that your review comment makes little sense as the `find_by!` that @saraycp patches out is doing exactly what you're suggesting - including the creation of the issue.

So custom error message is all that's required - either through render_error or raise. I prefer raise, but exceptions aren't everyone's darling.

Reply:
@vpereira I understand your point, the DRYer the better. But this is an exceptional case and  we just want to customize the message at this point for now.

I've force-pushed already to use `raise` instead of `render_error`, halfway between both ideas.

Reply:
> don't care either way. I'm just saying that your review comment makes little sense as the `find_by!` that @saraycp patches out is doing exactly what you're suggesting

it was a rhetoric question @coolo ;)

@saraycp I don't see why we should move away from the `bang` finder methods.. any reason?  I had the impression, based in past reviews, that it was the rule here.. BTW I think the action `accept` has the same problem right? So maybe a `before_action` taking care of that for both methods - raising something like https://github.com/openSUSE/open-build-service/blob/00d38f203dd9469c132a3abb3de00fb2525946ad/src/api/app/controllers/source/errors.rb would be, IMO,  a better approach

Reply:
I personally find the proposed solution pretty brilliant - no need for a different exception. And the bang method throws that error the issue is about - you can't pass that message to the exception as @saraycp does here, so I see no way not to move away from them *and* having custom errors.